08-components-lab/src/02-numberguessing.jsx

```diff
@@ -1,17 +1,66 @@
-/**
- * Lab 02:
- *
- * Implement a number guessing game component.
- *
- * The component should provide an input field for the user to guess a
- * number picked randomly by the computer (from 1 to 1,000).
- * When typing a number, the component should print "too low" or "too high" 
- * as a hint to the user.
- *
- * To make the game interesting, computer should sometimes cheat and print wrong messages.
- */
-
-import React from 'react';
-
-export default React.createClass({
-});
+/**
+ * Lab 02:
+ *
+ * Implement a number guessing game component.
+ *
+ * The component should provide an input field for the user to guess a
+ * number picked randomly by the computer (from 1 to 1,000).
+ * When typing a number, the component should print "too low" or "too high" 
+ * as a hint to the user.
+ *
+ * To make the game interesting, computer should sometimes cheat and print wrong messages.
+ */
+
+import React from 'react';
+
+export default class MultiInput extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            result: "start guessing...",
+            number: Math.floor(Math.random() * (1000 - 1)) + 1,
+        };
+    }
+
+    guess = (ev) => {
+        console.log("guessed " + ev.target.value);
+        var result = this.checkResult(ev.target.value);
+        console.log(result);
+        this.setState({ result: result });
+    };
+
+    checkResult = (number) => {
+        var liar = Math.floor(Math.random() * (10 - 0)) + 0;

```

Comment:
I like your idea to give random chance for pc to be a liar, however I suggest instead of using liar < 2 and a switch case, to use it as following:

```
if (number == this.state.number) {
  return "correct";
} 
      
if (number > this.state.number){
 return liar < 2 ? "lower" : "higher";
} else {
  return liar < 2 ? "higher" : "lower";
}
```

In such a way you can be sure that pc will say "lower" when the actual number is higher. In your switch case you can't be sure that pc is lying.